Description: Fix a memory allocation bug in the C implementation.
Forwarded: yes
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2017-02-25

--- a/install-mimic.c
+++ b/install-mimic.c
@@ -1,5 +1,5 @@
 /*-
- * Copyright (c) 2016  Peter Pentchev
+ * Copyright (c) 2016, 2017  Peter Pentchev
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -52,7 +52,7 @@
 static void
 version(void)
 {
-	puts("install-mimic 0.2.0");
+	puts("install-mimic 0.3.0");
 }
 
 static void
@@ -221,7 +221,7 @@
 
 			const size_t fullsz = last_dir_len + !has_slash +
 			    strlen(base) + 1;
-			fullbuf = realloc(fullbuf, sz);
+			fullbuf = realloc(fullbuf, fullsz);
 			if (fullbuf == NULL)
 				err(1, "Could not allocate %zu bytes", fullsz);
 			snprintf_check(fullbuf, fullsz, "%s%s%s",
